### ISSUE TITLE
SyntaxTrivia doesn't properly validate it's SyntaxTree

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#40773]
Consider the following code:

``` cs
var trivia = SyntaxFactory.SyntaxTrivia(SyntaxKind.WhitespaceTrivia, " ");
var location = trivia.GetLocation();
```

This code will throw a `NullReferenceException` because the implementation assumes there is a `SyntaxTree` available and there is not in this case. The code should be validating this and throwing a better exception type here.

**Comment by CyrusNajmabadi** [comment]
I thought hte design of syntax was that if something didn't have a tree it would synthesize one it belonged to.  Don't syntax noes work that way?  if so, shouldn't tokens/trivia?

**Comment by jaredpar** [comment]
That is how `SyntaxNode`s work but apparently not `SyntaxTrivia`. I constructed similar examples for `SyntaxNode` when I was doing nullable annotations and verified they all worked in scenarios like this. But the above trivially demonstrates that `SyntaxTrivia` null ref's here.

**Comment by CyrusNajmabadi** [comment]
Right.  I get that SyntaxTrivia null-refs here :)  My point is: we should cahnge that to just not null-ref and instead *succeed*.  We should *not* have it throw another exception.

Not: i don't feel strongly about that.  I'm in totaly agreement the existing behavior is busted.  I just slightly disagree on hte desired behavior.

**Comment by jaredpar** [comment]
I'm fine with either behavior too.
